File: qb-engineer-ui/src/app/features/inventory/models/place-bin-content-request.model.ts
import { BinContentStatus } from './bin-content-status.type';

export interface PlaceBinContentRequest {
  locationId: number;
  entityType: string;
  entityId: number;
  quantity: number;
  lotNumber?: string;
  jobId?: number;
  status: BinContentStatus;
  notes?: string;
}

File: qb-engineer-server/qb-engineer.core/Models/PlaceBinContentRequestModel.cs
using QBEngineer.Core.Enums;

namespace QBEngineer.Core.Models;

public record PlaceBinContentRequestModel(
    int LocationId,
    string EntityType,
    int EntityId,
    decimal Quantity,
    string? LotNumber,
    int? JobId,
    BinContentStatus Status,
    string? Notes);
